### ISSUE TITLE
To train DD-PPO (depth model) using Gibson 4+ and MP3D (train/val/test)

### ISSUE THREAD

**Issue by Shigangli** [facebookresearch/habitat-lab#322]
## ❓ Questions and Help
Hi,

I can run DD-PPO on a HPC machine now. I would like to reproduce the following results.

![image](https://user-images.githubusercontent.com/10457294/75929933-da007580-5e71-11ea-9679-ce414a38504c.png)

Now I merged <gibson/v1/train/content>​ and  <mp3d/v1/train/content> in to the same folder and point habitat-api to that. But I still have the following problems.

(1) Could you please give detailed instructions for how to generate the episodes of mp3d val and test scenes, and how to put them into the training dataset?

(2) To achieve the SPL scores (0.956 and 0.941), which Val and Test datasets should I use? Namely how to prepare the Val and Test datasets?

(3) Could you please also share the file <ddppo_pointnav.yaml> and other base yaml files included in <ddppo_pointnav.yaml> to train the heterogeneous environments?

**Comment by erikwijmans** [comment]
Hi, 

1) I used a variant of this script: https://gist.github.com/erikwijmans/1c596183fb8de1e9afd80088b6b5c115 to generate the pointnav episodes (generated 50k per scene).

2) All val and test scores are reported on Gibson, so use the Gibson val and test scenes, https://github.com/facebookresearch/habitat-api#task-datasets and https://evalai.cloudcv.org/web/challenges/challenge-page/254/leaderboard/837

3) The only thing you need to change in one of the base pointnav configs is the DATA_PATH (i.e. here: https://github.com/facebookresearch/habitat-api/blob/master/configs/tasks/pointnav_gibson.yaml#L31  The only changed needed for the https://github.com/facebookresearch/habitat-api/blob/master/habitat_baselines/config/pointnav/ddppo_pointnav.yaml file is to point to the new task config.

**Comment by erikwijmans** [comment]
The number of machines shouldn't effect the amount of GPU memory used.  DD-PPO uses PyTorch's DistributedDataParallel to accumulate gradients across different GPUs, so you may have better luck debugging that by tasking there.

To lower memory usage, you can change the parameter here: https://github.com/facebookresearch/habitat-api/blob/master/habitat_baselines/config/pointnav/ddppo_pointnav.yaml#L30 (less steps, less memory usage, shouldn't hurt final performance too much).  One other option is to use resnet18 instead of resnet50 -- for depth at least, my experience is that the larger CNNs do provide additional performance, but it is a small gain.

**Comment by Shigangli** [comment]
I can train DD-PPO using more machines now. I trained the model for 100 updates for a small scale test.

But I didn't see the SPL scores printed out. I also visualize the tensorboard files but there is only the curve for the reward value. The command I used is as follows.

<srun python -u -m habitat_baselines.run --exp-config habitat_baselines/config/pointnav/ddppo_pointnav.yaml --run-type train>

How can I generate the curve for the SPL scores?
